tests/e2e/clusterreset/clusterreset_test.go

```diff
@@ -0,0 +1,181 @@
+
+func Test_E2EClusterReset(t *testing.T) {
+	RegisterFailHandler(Fail)
+	flag.Parse()
+	RunSpecs(t, "Create ClusterReset Test Suite")
+}
+
+var (
+	kubeConfigFile  string
+	serverNodeNames []string
+	agentNodeNames  []string
+)
+
+var _ = Describe("Verify Create", Ordered, func() {
+	Context("Cluster :", func() {
+		It("Starts up with no issues", func() {
+			var err error
+			if *local {
+				serverNodeNames, agentNodeNames, err = e2e.CreateLocalCluster(*nodeOS, *serverCount, *agentCount)
+			} else {
+				serverNodeNames, agentNodeNames, err = e2e.CreateCluster(*nodeOS, *serverCount, *agentCount)
+			}
+			Expect(err).NotTo(HaveOccurred(), e2e.GetVagrantLog())
+			fmt.Println("CLUSTER CONFIG")
+			fmt.Println("OS:", *nodeOS)
+			fmt.Println("Server Nodes:", serverNodeNames)
+			fmt.Println("Agent Nodes:", agentNodeNames)
+			kubeConfigFile, err = e2e.GenKubeConfigFile(serverNodeNames[0])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(e2e.DockerLogin(kubeConfigFile, *ci)).To(Succeed())
+		})
+
+		It("Checks Node and Pod Status", func() {
+			fmt.Printf("\nFetching node status\n")
+			Eventually(func(g Gomega) {
+				nodes, err := e2e.ParseNodes(kubeConfigFile, false)
+				g.Expect(err).NotTo(HaveOccurred())
+				for _, node := range nodes {
+					g.Expect(node.Status).Should(Equal("Ready"))
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParseNodes(kubeConfigFile, true)
+
+			fmt.Printf("\nFetching Pods status\n")
+			Eventually(func(g Gomega) {
+				pods, err := e2e.ParsePods(kubeConfigFile, false)
+				g.Expect(err).NotTo(HaveOccurred())
+				for _, pod := range pods {
+					if strings.Contains(pod.Name, "helm-install") {
+						g.Expect(pod.Status).Should(Equal("Completed"), pod.Name)
+					} else {
+						g.Expect(pod.Status).Should(Equal("Running"), pod.Name)
+					}
+				}
+			}, "420s", "5s").Should(Succeed())
+			_, _ = e2e.ParsePods(kubeConfigFile, true)
+		})
+
+		It("Verifies ClusterReset Functionality", func() {
+			Eventually(func(g Gomega) {
+				for _, nodeName := range serverNodeNames {
+					if nodeName != "server-0" {
+						cmd := "sudo systemctl stop k3s"
+						_, err := e2e.RunCmdOnNode(cmd, nodeName)
+						Expect(err).NotTo(HaveOccurred())
+					}
+				}
+
+				cmd := "sudo systemctl stop k3s"
+				_, err := e2e.RunCmdOnNode(cmd, "server-0")
+				Expect(err).NotTo(HaveOccurred())
+
+				cmd = "sudo k3s server --cluster-reset"
+				res, err := e2e.RunCmdOnNode(cmd, "server-0")
+				Expect(err).NotTo(HaveOccurred())
+				Expect(res).Should(ContainSubstring("cluster-reset"))
+
+				cmd = "sudo systemctl start k3s"
+				_, err = e2e.RunCmdOnNode(cmd, "server-0")
+				Expect(err).NotTo(HaveOccurred())
+
+				fmt.Printf("\nFetching node status\n")
+				Eventually(func(g Gomega) {
+					nodes, err := e2e.ParseNodes(kubeConfigFile, false)
+					g.Expect(err).NotTo(HaveOccurred())
+					for _, node := range nodes {
+						g.Expect(node.Status).Should(Equal("Ready"))
```

Comment:
After the initial cluster-reset, only `server-0` should be `Ready` and other nodes should be `NotReady` after some time. I think that might be a safer check for cluster-reset functionality. Will probably have to increase the timeout of this check for that too.